internal/driver/driver_test.go

```diff
@@ -1020,34 +1018,221 @@ func TestTagFilter(t *testing.T) {
 	}
 }
 
+func TestIdentifyNumLabelUnits(t *testing.T) {
+	var tagFilterTests = []struct {
+		name               string
+		tagVals            []map[string][]int64
+		tagUnits           []map[string][]string
+		wantUnits          map[string]string
+		ignoreRX           string
+		wantIgnoreErrCount int
+	}{
+		{
+			"test1",
+			[]map[string][]int64{{"key1": {131072}, "key2": {128}}},
+			[]map[string][]string{{"key1": {"bytes"}, "key2": {"kilobytes"}}},
+			map[string]string{"key1": "bytes", "key2": "kilobytes"},
+			"",
+			0,
+		},
+		{
+			"test3",
+			[]map[string][]int64{{"key1": {8, 8}}},
+			[]map[string][]string{{"key1": {"bytes", "kilobytes"}}},
+			map[string]string{"key1": "bytes"},
+			`(For tag key1 used unit bytes, also encountered unit\(s\) kilobytes)`,
+			1,
+		},
+		{
+			"test4",
+			[]map[string][]int64{{"key1": {8}}, {"key1": {8}}},
+			[]map[string][]string{{"key1": {"bytes"}}, {"key1": {"kilobytes"}}},
+			map[string]string{"key1": "bytes"},
+			`(For tag key1 used unit bytes, also encountered unit\(s\) kilobytes)`,
+			1,
+		},
+		{
+			"test5",
+			[]map[string][]int64{{
+				"alignment": {8},
+				"request":   {8},
+				"bytes":     {8},
+			}},
+			[]map[string][]string{{
+				"alignment": {"seconds"},
+				"request":   {"minutes"},
+				"bytes":     {"hours"},
+			}},
+			map[string]string{
+				"alignment": "seconds",
+				"request":   "minutes",
+				"bytes":     "hours",
+			},
+			"",
+			0,
+		},
+	}
+	for _, test := range tagFilterTests {
+		p := profile.Profile{Sample: make([]*profile.Sample, len(test.tagVals))}
+		for i, numLabel := range test.tagVals {
+			s := profile.Sample{
+				NumLabel: numLabel,
+				NumUnit:  test.tagUnits[i],
+			}
+			p.Sample[i] = &s
+		}
+		testUI := &proftest.TestUI{T: t, IgnoreRx: test.ignoreRX}
+		units := identifyNumLabelUnits(&p, testUI)
+		for key, wantUnit := range test.wantUnits {
+			unit := units[key]
+			if wantUnit != unit {
+				t.Errorf("%s: for key %s, got unit %s, want unit %s", test.name, key, unit, wantUnit)
```

Comment:
Since you got the test.name field (which should be test.desc, see above), I'd recommend using t.Run() to execute the test cases, see https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks.